Modify downstream spectrometer support

rahmans1 opened the pull request:
**Work in Progress. Do not merge yet.**  The PR modifies the inner support clamps for the downstream magnet subcoils. The position references are changed to the upstream face to make it easier to compare with CAD. 

Some minor overlaps are introduced with collimator 5, 6A, 6B. Right now checking if it's simple precision error or difference in shape. If it's precision error, the support definition will be amended to match the boundary. 

If it's difference in shape for collimators (probably due to how collimator slots into support groove), each collimator will be handled with a separate pull request. Question: Since the position of the support has shifted, will the collimator dimension need to be projected? See picture below. Everything is in original position except for the inner support clamps.
![ShowOffset](https://user-images.githubusercontent.com/7409132/182443886-a33cdaee-9406-48f9-b3f8-6a0f39210146.PNG)

The outer supports (end plates for each subcoil) are not yet translated to reflect updated CAD. Most are probably unchanged except in subcoil 4. Thinking it would be better to separate it into another small PR.

Once these changes are ready, all the small PRs can be merged together pending final review.

rahmans1 committed:
Replacing the solids for downstream magnet subcoil support clamps

Instead of using multiple solids for each clamp, use a single solid per clamp. Also move the reference position from the middle of the solid to the upstream face (easier for direct comparisons with CAD).

rahmans1 committed:
Replacing the logical volumes for downstream magnet subcoil support clamps

Simplified nomenclature. Clamp index increase as one moves downstream.

rahmans1 committed:
Replacing physical placements for downstream magnet subcoil support clamps

All positions mentioned are now the position of the upstream face (Easier comparison with CAD).

rahmans1 committed:
Reinstating accidentally removed solid for downstream subcoil 1 conductor

rahmans1 committed:
Removed collinear vertices and adjusted precision at micrometer level to match outer boundary of coils with insulation as implemented

Collinear vertices were throwing warnings. So, they are removed. Does not affect shape.
Also azimuthal width of the support opening needed to match with coil outer boundary to avoid big overlaps.

rahmans1 committed:
Fixed typo: Missing quotation mark for string

rahmans1 committed:
Cosmetic change. Align loop ends.

chandabindu commented:
Thanks for this tedious work.
Collimators 5, and 6s are inside the magnetic field. Simple projections may not work.

I am confused why there would be any overlap between collimators 5 and 6s if you read them (the inner clamps and collimators) from the recent CAD file. Does this mean there are overlaps in the CAD file JLab gave us? If so, please let Dave kashy know.

chandabindu commented:
Hi Sakib, Dave and Dan,
Since Sakib has found many issues while importing the downstream CAD in our
simulation model, can we meet briefly to discuss his findings?
I am wide open today.
Please let us know your availability.
Best,
Chandan

On Tue, Aug 2, 2022 at 2:15 PM Sakib Rahman ***@***.***>
wrote:

> @rahmans1 <https://github.com/rahmans1> requested your review on: #562
> <https://github.com/JeffersonLab/remoll/pull/562> Modify downstream
> spectrometer support.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/JeffersonLab/remoll/pull/562#event-7112235424>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH2AK7PVU62BJHEB4KPMHFDVXFQUFANCNFSM55MCV37Q>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>


-- 
Regards,
chandan

chandabindu commented:
I'm not available today.
Tomorrow morning between 07:00 and 09:00 I can join a meeting.
Dave
________________________________
From: Chandan Ghosh ***@***.***>
Sent: Wednesday, August 3, 2022 11:08 AM
To: JeffersonLab/remoll ***@***.***>
Cc: JeffersonLab/remoll ***@***.***>; Review requested ***@***.***>; Sakib Rahman ***@***.***>; David Kashy ***@***.***>; Daniel Young ***@***.***>
Subject: [EXTERNAL] Re: [JeffersonLab/remoll] Modify downstream spectrometer support (PR #562)

Hi Sakib, Dave and Dan,
Since Sakib has found many issues while importing the downstream CAD in our simulation model, can we meet briefly to discuss his findings?
I am wide open today.
Please let us know your availability.
Best,
Chandan

On Tue, Aug 2, 2022 at 2:15 PM Sakib Rahman ***@***.******@***.***>> wrote:

@rahmans1<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_rahmans1&d=DwMFaQ&c=CJqEzB1piLOyyvZjb8YUQw&r=EYT5Wio7R4s18YncI8np1w&m=0_SISVn1Z0siqH9YSwPITtzE4z7YgWKFcc1p_7BACyXxoDdIMiaSUY7C8dKshw5C&s=5QCCjPq1FDcnc8d909XXd_wHpoYATERidH9wIzyYxHg&e=> requested your review on: #562<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_JeffersonLab_remoll_pull_562&d=DwMFaQ&c=CJqEzB1piLOyyvZjb8YUQw&r=EYT5Wio7R4s18YncI8np1w&m=0_SISVn1Z0siqH9YSwPITtzE4z7YgWKFcc1p_7BACyXxoDdIMiaSUY7C8dKshw5C&s=2kNMkWhVqc3VJHYtQz9R3prQkJt9KPLiMyYolNjonE0&e=> Modify downstream spectrometer support.

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_JeffersonLab_remoll_pull_562-23event-2D7112235424&d=DwMFaQ&c=CJqEzB1piLOyyvZjb8YUQw&r=EYT5Wio7R4s18YncI8np1w&m=0_SISVn1Z0siqH9YSwPITtzE4z7YgWKFcc1p_7BACyXxoDdIMiaSUY7C8dKshw5C&s=pV9jjpOXfRETzz6GAsZsx2eWoV_lhuiskxEOTaM5Ddo&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AH2AK7PVU62BJHEB4KPMHFDVXFQUFANCNFSM55MCV37Q&d=DwMFaQ&c=CJqEzB1piLOyyvZjb8YUQw&r=EYT5Wio7R4s18YncI8np1w&m=0_SISVn1Z0siqH9YSwPITtzE4z7YgWKFcc1p_7BACyXxoDdIMiaSUY7C8dKshw5C&s=zocuX2SWKXvX9hPHsn02fYELPPeEACf81z0pY2tCUDU&e=>.
You are receiving this because your review was requested.Message ID: ***@***.***>


--
Regards,
chandan

chandabindu commented:
Sakib, Can we meet and discuss this?


On Wed, Aug 3, 2022 at 11:27 AM David Kashy ***@***.***> wrote:

> I'm not available today.
> Tomorrow morning between 07:00 and 09:00 I can join a meeting.
> Dave
> ------------------------------
> *From:* Chandan Ghosh ***@***.***>
> *Sent:* Wednesday, August 3, 2022 11:08 AM
> *To:* JeffersonLab/remoll <
> ***@***.***>
> *Cc:* JeffersonLab/remoll ***@***.***>; Review requested <
> ***@***.***>; Sakib Rahman ***@***.***>;
> David Kashy ***@***.***>; Daniel Young ***@***.***>
> *Subject:* [EXTERNAL] Re: [JeffersonLab/remoll] Modify downstream
> spectrometer support (PR #562)
>
> Hi Sakib, Dave and Dan,
> Since Sakib has found many issues while importing the downstream CAD in
> our simulation model, can we meet briefly to discuss his findings?
> I am wide open today.
> Please let us know your availability.
> Best,
> Chandan
>
> On Tue, Aug 2, 2022 at 2:15 PM Sakib Rahman ***@***.***>
> wrote:
>
> @rahmans1
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_rahmans1&d=DwMFaQ&c=CJqEzB1piLOyyvZjb8YUQw&r=EYT5Wio7R4s18YncI8np1w&m=0_SISVn1Z0siqH9YSwPITtzE4z7YgWKFcc1p_7BACyXxoDdIMiaSUY7C8dKshw5C&s=5QCCjPq1FDcnc8d909XXd_wHpoYATERidH9wIzyYxHg&e=>
> requested your review on: #562
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_JeffersonLab_remoll_pull_562&d=DwMFaQ&c=CJqEzB1piLOyyvZjb8YUQw&r=EYT5Wio7R4s18YncI8np1w&m=0_SISVn1Z0siqH9YSwPITtzE4z7YgWKFcc1p_7BACyXxoDdIMiaSUY7C8dKshw5C&s=2kNMkWhVqc3VJHYtQz9R3prQkJt9KPLiMyYolNjonE0&e=>
> Modify downstream spectrometer support.
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_JeffersonLab_remoll_pull_562-23event-2D7112235424&d=DwMFaQ&c=CJqEzB1piLOyyvZjb8YUQw&r=EYT5Wio7R4s18YncI8np1w&m=0_SISVn1Z0siqH9YSwPITtzE4z7YgWKFcc1p_7BACyXxoDdIMiaSUY7C8dKshw5C&s=pV9jjpOXfRETzz6GAsZsx2eWoV_lhuiskxEOTaM5Ddo&e=>,
> or unsubscribe
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AH2AK7PVU62BJHEB4KPMHFDVXFQUFANCNFSM55MCV37Q&d=DwMFaQ&c=CJqEzB1piLOyyvZjb8YUQw&r=EYT5Wio7R4s18YncI8np1w&m=0_SISVn1Z0siqH9YSwPITtzE4z7YgWKFcc1p_7BACyXxoDdIMiaSUY7C8dKshw5C&s=zocuX2SWKXvX9hPHsn02fYELPPeEACf81z0pY2tCUDU&e=>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>
>
>
> --
> Regards,
> chandan
>


-- 
Regards,
chandan

rahmans1 committed:
Adapting the second clamp on subcoil 3 to avoid overlap with Collimator 5

Adapt second clamp on subcoil 3 to avoid overlap with Collimator 5 from latest CAD
